### PR TITLE
Fix paths in the ingress template and values file written by helm create

### PR DESCRIPTION
**Pull request by liuliqiang**
Signed-off-by: Liu Liqiang <liuliqiang@smartx.com>

**What this PR does / why we need it**:

Fix issue #6037 

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/chartutil/create.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ ingress:
   annotations: {}
     # kubernetes.io/ingress.class: nginx
     # kubernetes.io/tls-acme: "true"
-  path: /
+  paths: []
   hosts:
     - chart-example.local
```
Comment (hickeyma) [inline pkg/chartutil/create.go]: `paths: []`  should be part of `host`

File: pkg/chartutil/create.go [hunk 2]
```diff
@@ -150,7 +150,7 @@ const defaultIgnore = `# Patterns to ignore when building packages.
 
 const defaultIngress = `{{- if .Values.ingress.enabled -}}
 {{- $fullName := include "<CHARTNAME>.fullname" . -}}
-{{- $ingressPath := .Values.ingress.path -}}
+{{- $ingressPaths := .Values.ingress.paths -}}
```
Comment (hickeyma) [inline pkg/chartutil/create.go]: Should instead be: `{{- $svcPort := .Values.service.port -}}`

**Review by hickeyma — CHANGES_REQUESTED** [review]
@liuliqiang Thanks for pushing this PR. However, a lot has changed with ingress since PR #4703. I would suggest looking at the current version of ingress in Helm v2 and update this PR accordingly.

**Comment by liuliqiang** [comment]
@hickeyma thanks for your comment，i learn a lot from these comments, i will modify my patch and make it better. 😀

**Comment by liuliqiang** [comment]
hi @hickeyma  , i have check the master branch and update this patch，i hope i doesn't miss anything.

File: pkg/chartutil/create.go [hunk 3]
```diff
@@ -159,28 +160,30 @@ metadata:
 {{ include "<CHARTNAME>.labels" . | indent 4 }}
   {{- with .Values.ingress.annotations }}
   annotations:
-{{ toYaml . | indent 4 }}
-{{- end }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
```
Comment (hickeyma) [inline pkg/chartutil/create.go]: Decrease by 2 indents so `{{- end }}` is inline with `{{- with .Values.ingress.annotations }}`

**Review by hickeyma — CHANGES_REQUESTED** [review]
@liuliqiang Looking good.Just one small indentation nit. When you make the change, can you also squash the changes into one commit?

File: pkg/chartutil/create.go (hunk 3 above)
Comment (liuliqiang) [reply pkg/chartutil/create.go]: Ok，i will do it now，i am sorry for that。

**Comment by liuliqiang** [comment]
hi @hickeyma ，i have merge commits into one.
And also align the `{{- with .Values.ingress.annotations }}` and `{{end}}`

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @liuliqiang.

**Pull request merged**